src/plans/debug.jl

```diff
@@ -946,7 +948,59 @@ function (d::DebugWarnIfFieldNotFinite)(
     return nothing
 end
 function show(io::IO, dw::DebugWarnIfFieldNotFinite)
-    return print(io, "DebugWarnIfFieldNotFinite(:$(dw.field))")
+    return print(io, "DebugWarnIfFieldNotFinite(:$(dw.field), :$(dw.status))")
+end
+
+@doc raw"""
+    DebugWarnIfGradientNormTooLarge{T} <: DebugAction
+
+A debug to warn when an evaluated gradient at the current iterate is larger than
+(a factor times) the injectivity radius at the current iterate.
+
+# Constructor
+    DebugWarnIfGradientNormTooLarge(warn=:Once, factor::T=1.0)
+
+Initialize the warning to warn `:Once`.
+
+This can be set to `:Once` to only warn the first time the cost is Nan.
+It can also be set to `:No` to deactivate the warning, but this makes this Action also useless.
+All other symbols are handled as if they were `:Always:`
+
+# Example
+    DebugWaranIfFieldNotFinite(:Gradient)
+
+Creates a [`DebugAction`] to track whether the gradient does not get `Nan` or `Inf`.
+"""
+mutable struct DebugWarnIfGradientNormTooLarge{T} <: DebugAction
+    status::Symbol
+    factor::T
+    function DebugWarnIfGradientNormTooLarge(factor::T=1.0, warn::Symbol=:Once) where {T}
+        return new{T}(warn, factor)
+    end
+end
+function (d::DebugWarnIfGradientNormTooLarge)(
+    mp::AbstractManoptProblem, st::AbstractManoptSolverState, i::Int
+)
+    if d.status !== :No
+        M = get_manifold(mp)
+        p = get_iterate(st)
+        X = get_gradient(st)
+        Xn = norm(M, p, X)
+        p_inj = d.factor * injectivity_radius(M, p)
```

Comment:
Given that tangent bundle has 0 injectivity radius, I'd suggest using `max_stepsize` here instead of `injectivity_radius`.

Reply:
I think I do not understand what you mean here – I think the tangent bundle has even infectivity of infinite, since in every tangent space we have a vector space?

Besides that, you are right, maybe max_stepsize fits a bit better.

Reply:
Changed it to `max_stepsize`.

Reply:
Tangent *space* has infinite injectivity radius but tangent *bundle* (with the Sasaki metric) has injectivity radius 0 if the underlying manifold is not flat: https://mathoverflow.net/a/94348 .

Reply:
Sasaki Metric was the thing I missed here.